daml-lf/archive/src/stable/protobuf/com/daml/daml_lf_1_15/daml_lf_1.proto

```diff
@@ -0,0 +1,1808 @@
+  TEXT_TO_NUMERIC = 117;  // *Available in versions >= 1.7*
+  CONTRACT_ID_TO_TEXT = 136; // *Available in versions >= 1.11*
+  SHA256_TEXT = 93;
+
+  DATE_TO_UNIX_DAYS = 72; // Date -> Int64
+  UNIX_DAYS_TO_DATE = 73; // Int64 -> Date
+
+  TIMESTAMP_TO_UNIX_MICROSECONDS = 74; // Timestamp -> Int64
+  UNIX_MICROSECONDS_TO_TIMESTAMP = 75; // Int64 -> Timestamp
+
+  INT64_TO_DECIMAL = 76;  // *Available in versions < 1.7*
+  DECIMAL_TO_INT64 = 77;  // *Available in versions < 1.7*
+
+  INT64_TO_NUMERIC = 118;  // *Available in versions >= 1.7*
+  NUMERIC_TO_INT64 = 119;  // *Available in versions >= 1.7*
+
+  IMPLODE_TEXT = 78;
+
+  EQUAL_INT64 = 79;       // *Available in versions < 1.11*
+  EQUAL_DECIMAL = 80;     // *Available in versions < 1.7*
+  EQUAL_NUMERIC = 120;    // *Available in versions >= 1.7 and < 1.11*
+  EQUAL_TEXT = 81;        // *Available in versions < 1.11*
+  EQUAL_TIMESTAMP = 82;   // *Available in versions < 1.11*
+  EQUAL_DATE = 83;        // *Available in versions < 1.11*
+  EQUAL_PARTY = 84;       // *Available in versions < 1.11*
+  EQUAL_BOOL = 85;        // *Available in versions < 1.11*
+  EQUAL_CONTRACT_ID = 86; // *Available in versions < 1.11*
+  EQUAL_LIST = 87;
+  EQUAL_TYPE_REP = 123;   // *Available in versions = 1.8*
+
+  EQUAL = 131;            // *Available in versions >= 1.11*
+  LESS_EQ = 132;          // *Available in versions >= 1.11*
+  LESS = 133;             // *Available in versions >= 1.11*
+  GREATER_EQ = 134;       // *Available in versions >= 1.11*
+  GREATER = 135;          // *Available in versions >= 1.11*
+
+  TRACE = 88;
+
+  COERCE_CONTRACT_ID = 102;
+
+  CODE_POINTS_TO_TEXT = 105;
+  TEXT_TO_CODE_POINTS = 106;
+
+  SCALE_BIGNUMERIC = 137; // *Available in versions >= 1.13*
+  PRECISION_BIGNUMERIC = 138; // *Available in versions >= 1.13*
+  ADD_BIGNUMERIC = 139; // *Available in versions >= 1.13*
+  SUB_BIGNUMERIC = 140; // *Available in versions >= 1.13*
+  MUL_BIGNUMERIC = 141; // *Available in versions >= 1.13*
+  DIV_BIGNUMERIC = 142; // *Available in versions >= 1.13*
+  SHIFT_RIGHT_BIGNUMERIC = 143; // *Available in versions >= 1.13*
+  BIGNUMERIC_TO_NUMERIC = 144; // *Available in versions >= 1.13*
+  NUMERIC_TO_BIGNUMERIC = 145; // *Available in versions >= 1.13*
+  BIGNUMERIC_TO_TEXT = 146; // *Available in versions >= 1.13*
+
+  TYPEREP_TYCON_NAME = 148; // *Available in versions >= 1.dev*
+
+  // Next id is 149.
+
+  // EXPERIMENTAL TEXT PRIMITIVES -- these do not yet have stable numbers.
+  TEXT_TO_UPPER = 9901; // *Available in versions >= 1.dev*
+  TEXT_TO_LOWER = 9902; // *Available in versions >= 1.dev*
+  TEXT_SLICE = 9903; // *Available in versions >= 1.dev*
+  TEXT_SLICE_INDEX = 9904; // *Available in versions >= 1.dev*
+  TEXT_CONTAINS_ONLY = 9905; // *Available in versions >= 1.dev*
+  TEXT_REPLICATE = 9906; // *Available in versions >= 1.dev*
+  TEXT_SPLIT_ON = 9907; // *Available in versions >= 1.dev*
+  TEXT_INTERCALATE = 9908; // *Available in versions >= 1.dev*
```

Comment:
```suggestion
```

The EXPERIMENTAL TEXT PRIMITIVES aren't in the 1.14 file and I don't think we want to release them yet.

The "Next id" comment only really makes sense in the 1.dev file.